homeassistant/components/surepetcare/binary_sensor.py

```diff
@@ -41,32 +43,30 @@ async def async_setup_platform(
             EntityType.FEEDER,
             EntityType.FELAQUA,
         ]:
-            entities.append(DeviceConnectivity(surepy_entity.id, spc))
+            entities.append(DeviceConnectivity(surepy_entity.id, coordinator))
         elif surepy_entity.type == EntityType.PET:
-            entities.append(Pet(surepy_entity.id, spc))
+            entities.append(Pet(surepy_entity.id, coordinator))
         elif surepy_entity.type == EntityType.HUB:
-            entities.append(Hub(surepy_entity.id, spc))
+            entities.append(Hub(surepy_entity.id, coordinator))
 
-    async_add_entities(entities, True)
+    async_add_entities(entities)
 
 
-class SurePetcareBinarySensor(BinarySensorEntity):
+class SurePetcareBinarySensor(BinarySensorEntity, CoordinatorEntity):
```

Comment:
Please invert the inheritance order. Generally the platform specific entity should come last.